sdk/tests/platform/test_az_mqtt_policy.c

```diff
@@ -0,0 +1,254 @@
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <cmocka.h>
+
+void az_platform_critical_error(void) { assert_true(false); }
+
+// MQTT information for testing
+#define TEST_MQTT_ENDPOINT "test.mosquitto.org"
+#define TEST_MQTT_PORT 1884 // Unencrypted, authenticated port.
+#define TEST_MQTT_USERNAME "rw"
+#define TEST_MQTT_PASSWORD "readwrite"
+#define TEST_MQTT_CLIENT_ID "mqtt_client_id_0505"
+#define TEST_MQTT_TOPIC "test_topic"
+#define TEST_MQTT_PAYLOAD "test_payload"
+
+#define TEST_MAX_RESPONSE_CHECKS 3
+#define TEST_RESPONSE_DELAY_MS 500 // Response from endpoint is slow. Temporary.
+
+static az_mqtt test_mqtt_client;
+static _az_mqtt_policy test_mqtt_policy;
+static _az_hfsm test_inbound_hfsm;
+static _az_event_pipeline test_event_pipeline;
+
+// Set variables to track the number of times each event is called.
+static int ref_connack = 0;
+static int ref_suback = 0;
+static int ref_puback = 0;
+static int ref_recv = 0;
+static int ref_disconnect = 0;
+
+static az_mqtt_connect_data test_mqtt_connect_data = {
+  .host = AZ_SPAN_LITERAL_FROM_STR(TEST_MQTT_ENDPOINT),
+  .port = TEST_MQTT_PORT,
+  .username = AZ_SPAN_LITERAL_FROM_STR(TEST_MQTT_USERNAME),
+  .password = AZ_SPAN_LITERAL_FROM_STR(TEST_MQTT_PASSWORD),
+  .client_id = AZ_SPAN_LITERAL_FROM_STR(TEST_MQTT_CLIENT_ID),
+  .certificate = { .cert = AZ_SPAN_LITERAL_EMPTY, .key = AZ_SPAN_LITERAL_EMPTY, .key_type = 0 },
+};
+
+static az_event_policy_handler az_inbound_hfsm_get_parent(az_event_policy_handler child_handler)
+{
+  (void)child_handler;
+  return NULL;
+}
+
+static az_result test_inbound_hfsm_root(az_event_policy* me, az_event event)
+{
+  (void)me;
+  int ret = AZ_OK;
+
+  switch ((int)event.type)
+  {
+    case AZ_MQTT_EVENT_CONNECT_RSP:
+      ref_connack++;
+      break;
+
+    case AZ_MQTT_EVENT_SUBACK_RSP:
+      ref_suback++;
+      break;
+
+    case AZ_MQTT_EVENT_PUBACK_RSP:
+      ref_puback++;
+      break;
+
+    case AZ_MQTT_EVENT_PUB_RECV_IND:
+      ref_recv++;
+      break;
+
+    case AZ_MQTT_EVENT_DISCONNECT_RSP:
+      ref_disconnect++;
+      break;
+  }
+
+  return ret;
+}
+
+static void test_az_mqtt_policy_init(void** state)
+{
+  (void)state;
+
+  az_context test_context;
+
+  assert_int_equal(
+      _az_hfsm_init(
+          &test_inbound_hfsm,
+          test_inbound_hfsm_root,
+          az_inbound_hfsm_get_parent,
+          (az_event_policy*)&test_mqtt_policy,
+          NULL),
+      AZ_OK);
+
+  assert_int_equal(
+      _az_mqtt_policy_init(
+          &test_mqtt_policy,
+          &test_mqtt_client,
+          &test_context,
+          NULL,
+          (az_event_policy*)&test_inbound_hfsm),
+      AZ_OK);
+
+  assert_int_equal(
+      _az_event_pipeline_init(
+          &test_event_pipeline,
+          (az_event_policy*)&test_mqtt_policy,
+          (az_event_policy*)&test_inbound_hfsm),
+      AZ_OK);
+}
+
+#ifndef AZ_NO_PRECONDITION_CHECKING
+
+ENABLE_PRECONDITION_CHECK_TESTS()
+
+static void test_az_mqtt_init_null(void** state)
```

Comment:
Please rename all tests to use our UT convention: e.g.
`test_az_iot_provisioning_client_get_connect_info_insufficient_space_fails`

test_object_action_expectedresult